cheatsheets/SAML_Security_Cheat_Sheet.md

```diff
@@ -106,14 +106,128 @@ Unsolicited Response is inherently [less secure](https://www.identityserver.com/
+
+It is worth noting that the CABF does not have governance around the use or acquisition of SAML certificates, certificates from their member CAs are what are widely considered Public CAs. That is, they are widely trusted by browsers, operating systems, and various development frameworks.
+
+Using Public CA signed certificates allows for revocation checking, which can increase security, but if the certificate exchange is not secured, this could lead to a false sense of security.
+
+##### Private CA Signed
+
+As most IdPs and SPs treat the X.509 certificates as an explicit trust, private CAs and PKI could be used. How private CAs are designed, built, and run varies wildly and ultimately running CAs well is very costly. In order to trust a third-party's CAs, one would need to clearly understand the lifecycle of the CA. There are two audit types that would cover this, both of which are very costly, on top of building and running the CAs. If you rely on third-party CAs, they should be [WebTrust](http://www.webtrust.org/), [ETSI](http://www.etsi.org/technologies-clusters/technologies/security/certification-authorities-and-other-certification-service-providers), or [SOC 2 Type II](https://www.aicpa-cima.com/topic/audit-assurance/audit-and-assurance-greater-than-soc-2) audited.
+
+Trusting third-party CAs, if done improperly, could result in unintended over trust, for things such as TLS and code signing. If you choose to trust third-party CAs, make sure they are only trusted for the process of IdP signature validation.
+
+If third-party CAs are used they still should not issue SAML signing certificates where the lifetime of the certificate exceeds that of the underlying key pair, based on guidance from a standards organization such as [NIST, NSA, etc.](https://www.keylength.com/en/). If using the strongest private key types, this puts the upper limit at two years.
+
+##### Self-Signed
+
+Due to the explicit nature of most SAML security associations, self-signed certificates are ideal for the use case. The contents of the certificate and lifetime are not constrained by the policy or process of the issuing CA, be it public or private. As rotating SAML certificates can be painful and labor intensive, setting the certificate lifetime as long as safely possible is key. Few CAs allow long enough lifetimes, due to their focus on the TLS threat model.
+
+###### Creating a Self-Signed SAML Certificate
+
+If you are using a Hardware Security Module (HSM), follow the vendor's instructions. This process uses openssl.
+
+1. Generate a Private Key:
+openssl genrsa -out private.key 2048
+
+2. Create a Configuration File (e.g., cert.cnf):
+
+\[req\]
+distinguished_name = req_distinguished_name
+x509_extensions = v3_ca
+prompt = no
+
+\[req_distinguished_name\]
+C = US
+ST = California
+L = San Francisco
+O = MyOrganization
+OU = MyUnit
+CN = SAML Signing
```

Comment:
Some companies, especially during transition, might be moving off FOSS (e.g., Shibboleth) to a COTS solution (e.g., Okta, Ping Identity, etc.), so in those cases there probably should be separate certs and thus different (i.e., more specific) CNs.